content/2_Bronze/Intro_Graphs.mdx

```diff
@@ -71,6 +71,316 @@ categories:
 
 Knowing [DFS](/silver/dfs) can be helpful but it should not be required.
 
+## Example - Grass Planting
+
+<FocusProblem problem="tutorial" />
+
+<Spoiler title="Hint 1">
+Consider a graph with a single center node, and some number of nodes adjacent to it:
+
+![Star Graph](./assets/graphfocushint.png)
+
+How many grass types do we need for this graph? How can we generalize to a larger graph?
+</Spoiler>
+<Spoiler title="Intended Solution">
+Let $\texttt{deg}[i]$ represent the degree of node $i$: the number of paths
+connecting the node. Then, considering that node, and that node alone, you would
+need $\texttt{deg}[i]+1$ grass types. This is because you would need
+$\texttt{deg}[i]$ for each of the adjacent nodes, and $1$ for the node itself.
+It can be shown that the tree can also be colored accordingly in
+$\max(\texttt{deg}[i])+1$ colors.
+
+<LanguageSection>
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+// FastIO: see General -> Fast Input and Output
+void setIO(string name = "") {
+	ios_base::sync_with_stdio(0);
+	cin.tie(0);
+	if (!name.empty()) {
+		freopen((name + ".in").c_str(), "r", stdin);
+		freopen((name + ".out").c_str(), "w", stdout);
+	}
+}
+
+int main() {
+	setIO("planting");
+
+	int field_num;
+	cin >> field_num;
+
+	vector<int> deg(field_num + 1);  // 1-indexed
+	for (int p = 0; p < field_num - 1; p++) {
+		int field1, field2;
+		cin >> field1 >> field2;
+
+		deg[field1]++;
+		deg[field2]++;
+	}
+
+	int max_deg = 0;
+	for (int f = 1; f <= field_num; f++) { max_deg = max(max_deg, deg[f]); }
+
+	cout << max_deg + 1 << endl;
+}
+```
+
+</CPPSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.StringTokenizer;
+
+public class Planting {
+	public static void main(String[] args) throws IOException {
+		Kattio io = new Kattio("planting");
+
+		int fieldNum = io.nextInt();
+		int[] deg = new int[fieldNum + 1];  // 1-indexed
+		for (int p = 0; p < fieldNum - 1; p++) {
+			int field1 = io.nextInt();
+			int field2 = io.nextInt();
+			deg[field1]++;
+			deg[field2]++;
+		}
+
+		int maxDeg = 0;
+		for(int f = 1; f <= fieldNum; f++) {
+			maxDeg = Math.max(maxDeg, deg[f]);
+		}
+
+		io.println(maxDeg + 1);
+		io.close();
+	}
+
+	//CodeSnip{Kattio}
+}
+```
+
+</JavaSection>
+<PySection>
+
+```py
+with open("planting.in") as read:
+	field_num = int(read.readline())
+	deg = [0 for _ in range(field_num + 1)]  # 1-indexed
+	for _ in range(field_num - 1):
+		field1, field2 = [int(i) for i in read.readline().split()]
+		deg[field1] += 1
+		deg[field2] += 1
+
+max_deg = max(deg)
+print(max_deg + 1, file=open("planting.out", "w"))
+```
+
+</PySection>
+</LanguageSection>
+</Spoiler>
+
+
+<Spoiler title="Alternate Solution">
+**Time Complexity**: $\mathcal{O}(N)$
+
+This can also be solved using [DFS](/silver/dfs).
```

Comment:
this problem probably shouldn't be in the Bronze module

Reply:
ack ok